src/sentry/search/events/builder/metrics.py

```diff
@@ -153,6 +153,9 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
                     "Must include on demand metrics type when querying on demand"
                 )
 
+            # Instead of calling OnDemandMetricSpec without a spec_version (defaulting to the least),
+            # we keep this logic here to make future spec versions easier to roll out since it will
+            # save us to look where to add this logic again
```

Comment:
@iambriccardo Does this read well?

I had to close #63279 because it was a mess to fix the rebasing or merging master.